ow_faults/src/FaultInjector.cpp

```diff
@@ -75,16 +76,54 @@ void FaultInjector::setFaultsMessage(ow_faults::PTFaults& msg, ComponentFaults v
   msg.value = static_cast<uint>(value); //should be HARDWARE for now
 }
 
-void FaultInjector::setPowerTemperatureFaultValue(bool getTempBool){
-  float thermal_val;
-  if (isnan(powerTemperatureOverloadValue)) {
-    thermal_val =  50.0 + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(90.0-50.0)));
-    powerTemperatureOverloadValue = thermal_val;
-  } else if (!getTempBool) {
-    powerTemperatureOverloadValue = NAN;
+float FaultInjector::getRandomValueFromRange( float min_val, float max_val){
+  float randomValue = min_val + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(max_val-min_val)));
+  return randomValue;
```

Comment:
This formula is not correct. The term `max_val-min_val` should be multiplied by the term `rand/rand_max` not divided! + you may simplify things a bit as follow
```suggestion
  return min_val + (max_val - min_val) * (rand() / float(RAND_MAX));
```

Reply:
updated and can confirm still works as expected.